src/buildvars.py

```diff
@@ -124,6 +124,9 @@ def _update_remote_bvars(stackname, buildvars):
 def refresh(stackname, context):
     "(safely) replaces the buildvars file on the ec2 instance(s)"
 
+    from master import download_keypair
+    download_keypair(stackname)
```

Comment:
so this hack has been hanging around in my git stash for a while now and needed to be committed. obviously it's not it's final form, but it solves an issue where you run `update_infrastructure` on a stack that your builder installation hasn't visited before.

Reply:
`download_keypair` seems like a safe operation for an existing stack. Move up the `import` to the top?

Reply:
Not sure why it is in `master` however, it's generic for any stack that has an `ec2` configuration.